addons/mail/controllers/attachment.py

```diff
@@ -102,8 +99,7 @@ def mail_attachment_delete(self, attachment_id, access_token=None, **kwargs):
         else:
             if (
                 not access_token
-                or not attachment_sudo.access_token
-                or not consteq(access_token, attachment_sudo.access_token)
+                or not verify_limited_field_access_token(attachment_sudo, "raw", access_token)
```

Comment:
Now that we have these tools, we could be able to clean this part (and the condition + comment above) if we provided a "delete" limited token (only to the author) rather than using the actual "display" token to verify access.